cbpfc.go

```diff
@@ -467,17 +477,25 @@ func addIndirectPacketGuards(blocks []*block) {
 }
 
 type packetGuardOpts struct {
-	// requiredGuard returns:
-	// - the packetGuard needed by insns
-	// - the number of instructions in insns covered by the guard.
-	//   The guard is assumed to be invalidated for the remaining / uncovered insns (eg RegX was clobbered for indirect guards).
-	//   requiredGuard will be called until all instructions are covered.
-	requiredGuard func(insns []instruction) (int, packetGuard)
+	// requiredGuard returns the packetGuard needed by insns, and what insns it is valid for.
+	requiredGuard func(insns []instruction) requiredGuard
 
 	// createInsn creates an instruction that checks the packet length against the guard
 	createInsn func(guard packetGuard) bpf.Instruction
 }
 
+type requiredGuard struct {
+	guard packetGuard
```

Comment:
Nit: maybe `offset`?

Reply:
That might have been a better name but we call it `guard` in a bunch of places already :(